### ISSUE TITLE
Should we recommend shoulda matchers for testing validations?

### ISSUE THREAD

**Issue by Darhazer** [rubocop-hq/rspec-style-guide#33]
https://github.com/thoughtbot/shoulda-matchers makes attribute validations much more readable and avoid some boilerplate.

For example, the current suggestion is
```ruby
describe '#title' do
   it 'is required' do
     article.title = nil
     article.valid?
     expect(article.errors[:title].size).to eq(1)
  end
end
 ```

while with shoulda_matchers it would looks like
```ruby
describe '#title' do
   it 'is required' do
     expect(article).to validate_presence_of(:title)
  end
end
 ```

**Comment by pirj** [comment]
WDYT of splitting this into two parts:
- Factor out commonly used logic into matchers
- Use 

I think it's quite fair to recommend `shoulda-matchers`, it's about to hit a major release, and it's *the* most known third-party RSpec matcher gem since we do already recommend FactoryBot and TimeCop.

**Comment by bbatsov** [comment]
I agree with @pirj on both counts. 

In general when it comes to third-party libs we should probably mention that we're recommending some extensions because they are quite popular, but generally any similar tool would do. The underlying principles are what really matters, not some concrete implementation of them.

**Comment by pirj** [comment]
Fixed by #49 and #50.
